include/deal.II/matrix_free/operators.h

```diff
@@ -1028,7 +1025,7 @@ namespace MatrixFreeOperators
     Assert((Base<dim, Number>::data != NULL), ExcNotInitialized());
 
     unsigned int dummy = 0;
-    LinearAlgebra::distributed::Vector<Number> &inverse_diagonal_entries = Base<dim,Number>::inverse_diagonal_entries;
+    LinearAlgebra::distributed::Vector<Number> inverse_diagonal_entries;
```

Comment:
You could still keep the variable `inverse_diagonal_entries` here if you do `this->inverse_diagonal_entries.get_vector()`. That way you avoid copying the vector and holding two vectors for a little while. (The same above for the mass matrix.)